util/fipstools/inject_hash/macho_parser/macho_parser.h

```diff
@@ -0,0 +1,46 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR ISC
+
+#ifndef MACHO_PARSER_H
+#define MACHO_PARSER_H
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+#include <mach-o/loader.h>
+#include <mach-o/nlist.h>
+
+typedef struct {
+    char name[16];
+    size_t size;
+    uint32_t offset;
+} section_info;
+
+// Since we only support 64-bit architectures on Apple, we don't need to account for any of the 32-bit structures
+#define LC_SEG LC_SEGMENT_64
+#define BIT_MODIFIER 8
+
+typedef struct mach_header_64 macho_header;
```

Comment:
I'm not a fan of using `typedef` to alias library-provided types to custom names. Maintaining/scaling this code might be easier by leaving the type names unchanged.